pyvlx/api/api_event.py

```diff
@@ -23,12 +23,18 @@ def __init__(self, pyvlx: "PyVLX", timeout_in_seconds: int = 10):
 
     async def do_api_call(self) -> None:
         """Start. Sending and waiting for answer."""
-        self.pyvlx.connection.register_frame_received_cb(self.response_rec_callback)
-        await self.send_frame()
-        await self.start_timeout()
-        await self.response_received_or_timeout.wait()
-        await self.stop_timeout()
-        self.pyvlx.connection.unregister_frame_received_cb(self.response_rec_callback)
+        # We check for connection before entering the semaphore section
+        # because otherwise we might try to connect, which calls this, and we get stuck on
+        # the semaphore.
+        await self.pyvlx.check_connected()
```

Comment:
Why you exclude check_connected() from Semaphore?

Reply:
Less a matter of excluding, more a matter of "do before". So, inside `check_connected` is sometimes (if not connected yet) a call to `connect`, which calls `self.klf200.password_enter` which calls `do_api_call`...

Without checking for connection before, the first call to `do_api_call` (a `GetAllNodesInformation` typically) gets stuck because the semaphore has been activated, but it can't connect.

The magic that makes this not loop entirely is that the first thing `connect` does is call `await self.connection.connect()` so when `check_connected` gets called as part of the `password_enter` `do_api_call`, it doesn't call `connect`.

Reply:
Ok, you'right, I don't mean exclude, but before Semaphore. Especially, why not checking connection "inside" Semaphore?

What happen if you receive a burst of commands from HA while the pyvlx is not connected? Don't you trigger the connect function several times? Why not making this check within the Semaphore?